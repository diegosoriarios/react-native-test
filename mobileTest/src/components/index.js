export { default as CustomButton } from './CustomButton'
export { default as CustomInput } from './CustomInput'
export { default as Form } from './Form'
export { default as GridItem } from './GridItem'
export { default as IconLink} from './IconLink'
export { default as ImageChoose} from './ImageChoose'
export { default as ListItem} from './ListItem'
export { default as Loading} from './Loading'