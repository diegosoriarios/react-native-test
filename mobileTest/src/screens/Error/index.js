import React from 'react';
import { View, Text } from 'react-native';

// import { Container } from './styles';

const Error = () => {
  return (
    <View>
      <Text>Something happened</Text>
    </View>
  )
}

export default Error;