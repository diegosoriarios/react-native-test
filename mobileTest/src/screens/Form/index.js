import React, { useState } from 'react';
import { View, Text, TouchableHighlight, StyleSheet } from 'react-native';
import CustomInput from '../../components/CustomInput/index';

// import { Container } from './styles';
import { connect } from 'react-redux';
import ImageChoose from '../../components/ImageChoose/index';
import CustomButton from '../../components/CustomButton/index';
import { myUserUpdate } from '../../actions/usersActions';
import { useNavigation } from '@react-navigation/native';

const Form = ({ myUser, updateCurrentUser, buttonText = "Login" }) => {

  const [name, setName] = useState(myUser.name)
  const [email, setEmail] = useState(myUser.email)
  const [description, setDescription] = useState(myUser.description)
  const [avatar, setAvatar] = useState(myUser.avatar)
  const navigation = useNavigation()

  const updateUser = () => {
    const userUpdated = {name, email, description, avatar}
    updateCurrentUser(userUpdated)
    navigation.navigate("Home")
  }

  return (
    <View style={styles.container}>
      <ImageChoose 
        image={avatar}
        onChangeImage={() => {}}
      />
      <CustomInput 
        placeholder="Name"
        value={name}
        onChangeText={setName}
      />
      <CustomInput 
        placeholder="E-mail"
        value={email}
        onChangeText={setEmail}
      />
      <CustomInput 
        placeholder="Description" 
        value={description}
        onChangeText={setDescription}
        line={4}
      />
      <CustomButton
        text={buttonText}
        onPress={updateUser}
      />
    </View>
  )
}

const mapStateToProps = state => {
  return {
    myUser: state.usersReducer.myUser
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateCurrentUser: (user) => dispatch(myUserUpdate(user))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Form);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: 'white'
  }
})