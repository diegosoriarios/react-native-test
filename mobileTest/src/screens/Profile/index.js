import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { FlatList, ActivityIndicator, View, Text, Image } from 'react-native';
import { SocialIcon } from 'react-native-elements'
import { useNavigation } from '@react-navigation/native';
import { getUserImages } from '../../actions/usersActions';
import styles from './styles'
import GridItem from '../../components/GridItem';
import Icon from 'react-native-vector-icons/FontAwesome'
Icon.loadFont();

const Profile = ({ route, imageIsLoading, getUserImages }) => {
  const { user } = route.params 
  const navigation = useNavigation()
  const [page, setPage] = useState(1)
  
  useEffect(() => {
    navigation.setOptions({ title: `${user.name}` })
    getUserImages(user.id, (user.id - 1) * 10 + page)
  }, [])  

  const getMorePhotos = () => {
    setPage(page + 1)
    getUserImages(user.id, (user.id - 1) * 10 + page)
  }

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Image style={styles.image} source={{ uri: user.image }} />
        <Text style={styles.textName}>{user.name}</Text>
        <Text style={styles.textEmail}>{user.email.toLowerCase()}</Text>
        <Text style={styles.textDescription}>{user.company.catchPhrase}</Text>
      </View>
      <View style={styles.socialDisplay}>
        {
          user.social.map((social, i) => (
             <SocialIcon key={i} type={social} />
          ))
        }
      </View>
      <View style={styles.list}>
        <FlatList
          data={user.albums}
          numColumns={3}
          keyExtractor={image => String(image.id)}
          ListFooterComponent={imageIsLoading && <ActivityIndicator size={'small'} color="#fff" />}
          renderItem={({item}) => <GridItem item={item} />}
          onEndReached={() => getMorePhotos()}
          onEndReachedThreshold={.3}
        />
      </View>
    </View>
  );
}

const mapStateToProps = state => {
  return {
    imageIsLoading: state.usersReducer.imageIsLoading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getUserImages: (id, page) => dispatch(getUserImages(id, page))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile);