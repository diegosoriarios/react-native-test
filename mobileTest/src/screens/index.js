export { default as HomeScreen } from './Home'
export { default as ProfileScreen } from './Profile'
export { default as Form } from './Form'
export { default as Error } from './Error' 