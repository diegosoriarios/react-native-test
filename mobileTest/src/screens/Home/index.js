import React, { useEffect } from 'react';
import { View, Text, FlatList } from 'react-native';
import { connect } from 'react-redux'

// import { Container } from './styles';
import { userFetchData } from '../../actions/usersFunctions';
import { ListItem } from '../../components';

const Home = ({ users, isLoading, hasErrored, fetchData }) => {

  useEffect(() => {
    fetchData()
  }, [])

  if (isLoading) return <Text>Loading</Text>
  if (hasErrored) return <Text>Error</Text>
  
  return (
    <FlatList 
      style={{ backgroundColor: 'white' }}
      data={users}
      keyExtractor={user => String(user.id)}
      renderItem={({item}) => <ListItem user={item} />}
    />
  )
}

const mapStateToProps = state => {
  return {
    users: state.usersReducer.users,
    isLoading: state.usersReducer.isLoading,
    hasErrored: state.usersReducer.hasErrored
  }
}

const mapDispatchToProps = dispatch => {
  return {
    fetchData: () => dispatch(userFetchData())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);