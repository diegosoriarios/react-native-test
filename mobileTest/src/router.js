import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { HomeScreen, ProfileScreen, Form, Error } from './screens'
import IconLink from './components/IconLink/index';
import Loading from './components/Loading/index';
import { navigationOptions } from './utils/navigationOptions';

const Stack = createStackNavigator();

function Router() {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} options={{
          headerStyle: navigationOptions.noHeader,
          headerLeft: () => <IconLink iconName={"settings"} page="Settings" />
        }} />
        <Stack.Screen name="Settings" component={Form} options={{
          headerStyle: navigationOptions.noHeader,
        }}/>
        <Stack.Screen name="Profile" component={ProfileScreen} options={{
          headerStyle: navigationOptions.noHeader,
          headerRight: () => <IconLink iconName="message" page="Chat" />
        }} />
        <Stack.Screen name="Chat" component={Loading} options={{
          headerStyle: navigationOptions.noHeader,
        }} />
        <Stack.Screen name="Error" component={Error} options={{
          headerStyle: navigationOptions.noHeader,
        }} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default Router;