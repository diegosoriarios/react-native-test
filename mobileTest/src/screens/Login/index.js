import React, { useState } from 'react'
import { useNavigation } from '@react-navigation/native';
import { connect } from 'react-redux';
import Form from '../../components/Form'
import { myUserUpdate } from '../../actions/usersActions';

function Login({ myUser, updateCurrentUser }) {
  const navigation = useNavigation()
  
  const [name, setName] = useState(myUser.name)
  const [email, setEmail] = useState(myUser.email)
  const [description, setDescription] = useState(myUser.description)
  const [avatar, setAvatar] = useState(myUser.avatar)

  const updateUser = () => {
    if (name.length && email.length) {
      const userUpdated = {name, email, description, avatar}

      updateCurrentUser(userUpdated)
      navigation.replace("Home") 
    } else {
      alert('Name and email cannot be null')
    }
  }

  const emailLowerCase = text => {
    setEmail(text.toLowerCase())
  }

  return (
    <Form 
      buttonText="Login" 
      updateUser={updateUser} 
      name={name}
      setName={setName}
      email={email}
      setEmail={emailLowerCase}
      description={description}
      setDescription={setDescription}
      avatar={avatar}
      setAvatar={setAvatar}
    />
  )
}

const mapStateToProps = state => {
  return {
    myUser: state.usersReducer.myUser
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateCurrentUser: (user) => dispatch(myUserUpdate(user))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)