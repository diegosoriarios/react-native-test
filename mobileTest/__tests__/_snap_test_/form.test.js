import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Form } from '../../src/screens'
import renderer from 'react-test-renderer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('@react-navigation/native');
test('Home Renders Correctly', () => {

  const store = mockStore({
    usersReducer: {
      myUser: {
        name: 'Diego',
        email: 'email',
        description: 'alskjfalsk'
      }
    }
  });

  const tree = renderer.create(
    <Provider store={store}>
      <Form />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});