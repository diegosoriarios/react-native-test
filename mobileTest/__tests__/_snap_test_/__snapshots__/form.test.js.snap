// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders Correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 75,
          "height": 150,
          "margin": 25,
          "width": 150,
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={1}
    onChangeText={[Function]}
    placeholder="Name"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FBE4F3",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginVertical": 10,
        "padding": 10,
        "textAlign": "center",
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value="Diego"
  />
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={1}
    onChangeText={[Function]}
    placeholder="E-mail"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FBE4F3",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginVertical": 10,
        "padding": 10,
        "textAlign": "center",
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value="email"
  />
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Description"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FBE4F3",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginVertical": 10,
        "padding": 10,
        "textAlign": "center",
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value="alskjfalsk"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#593CA1",
        "borderRadius": 50,
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 50,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FBE4F3",
          "fontWeight": "bold",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
